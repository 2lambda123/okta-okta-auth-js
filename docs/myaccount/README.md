@okta/okta-auth-js/myaccount / [Exports](modules.md)
