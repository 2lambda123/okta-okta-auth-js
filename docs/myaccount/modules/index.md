[@okta/okta-auth-js/myaccount](../README.md) / [Modules](../modules.md) / index

# Module: index

## Table of contents

### References

- [AddEmailPayload](index.md#addemailpayload)
- [AddPhonePayload](index.md#addphonepayload)
- [ChallengePhonePayload](index.md#challengephonepayload)
- [EmailProfile](index.md#emailprofile)
- [EmailRole](index.md#emailrole)
- [IAPIFunction](index.md#iapifunction)
- [MyAccountRequestOptions](index.md#myaccountrequestoptions)
- [PhoneProfile](index.md#phoneprofile)
- [Status](index.md#status)
- [UpdateProfilePayload](index.md#updateprofilepayload)
- [VerificationPayload](index.md#verificationpayload)

### Classes

- [BaseTransaction](../classes/index.BaseTransaction.md)
- [EmailChallengeTransaction](../classes/index.EmailChallengeTransaction.md)
- [EmailStatusTransaction](../classes/index.EmailStatusTransaction.md)
- [EmailTransaction](../classes/index.EmailTransaction.md)
- [PhoneTransaction](../classes/index.PhoneTransaction.md)
- [ProfileSchemaTransaction](../classes/index.ProfileSchemaTransaction.md)
- [ProfileTransaction](../classes/index.ProfileTransaction.md)

### Functions

- [addEmail](index.md#addemail)
- [addPhone](index.md#addphone)
- [deleteEmail](index.md#deleteemail)
- [deletePhone](index.md#deletephone)
- [getEmail](index.md#getemail)
- [getEmailChallenge](index.md#getemailchallenge)
- [getEmails](index.md#getemails)
- [getPhone](index.md#getphone)
- [getPhones](index.md#getphones)
- [getProfile](index.md#getprofile)
- [getProfileSchema](index.md#getprofileschema)
- [sendEmailChallenge](index.md#sendemailchallenge)
- [sendPhoneChallenge](index.md#sendphonechallenge)
- [updateProfile](index.md#updateprofile)
- [verifyEmailChallenge](index.md#verifyemailchallenge)
- [verifyPhoneChallenge](index.md#verifyphonechallenge)

## References

### AddEmailPayload

Re-exports [AddEmailPayload](types.md#addemailpayload)

___

### AddPhonePayload

Re-exports [AddPhonePayload](types.md#addphonepayload)

___

### ChallengePhonePayload

Re-exports [ChallengePhonePayload](types.md#challengephonepayload)

___

### EmailProfile

Re-exports [EmailProfile](types.md#emailprofile)

___

### EmailRole

Re-exports [EmailRole](../enums/types.EmailRole.md)

___

### IAPIFunction

Re-exports [IAPIFunction](types.md#iapifunction)

___

### MyAccountRequestOptions

Re-exports [MyAccountRequestOptions](types.md#myaccountrequestoptions)

___

### PhoneProfile

Re-exports [PhoneProfile](types.md#phoneprofile)

___

### Status

Re-exports [Status](../enums/types.Status.md)

___

### UpdateProfilePayload

Re-exports [UpdateProfilePayload](types.md#updateprofilepayload)

___

### VerificationPayload

Re-exports [VerificationPayload](types.md#verificationpayload)

## Functions

### addEmail

▸ **addEmail**(`oktaAuth`, `options?`): `Promise`<[`EmailTransaction`](../classes/index.EmailTransaction.md)\>

**`scope:`** okta.myaccount.email.manage

#### Parameters

| Name | Type |
| :------ | :------ |
| `oktaAuth` | `OktaAuthInterface` |
| `options?` | [`MyAccountRequestOptions`](types.md#myaccountrequestoptions) |

#### Returns

`Promise`<[`EmailTransaction`](../classes/index.EmailTransaction.md)\>

#### Defined in

[emailApi.ts:45](https://github.com/okta/okta-auth-js/blob/master/lib/myaccount/emailApi.ts#L45)

___

### addPhone

▸ **addPhone**(`oktaAuth`, `options?`): `Promise`<[`PhoneTransaction`](../classes/index.PhoneTransaction.md)\>

**`scope:`** okta.myaccount.phone.manage

#### Parameters

| Name | Type |
| :------ | :------ |
| `oktaAuth` | `OktaAuthInterface` |
| `options?` | [`MyAccountRequestOptions`](types.md#myaccountrequestoptions) |

#### Returns

`Promise`<[`PhoneTransaction`](../classes/index.PhoneTransaction.md)\>

#### Defined in

[phoneApi.ts:44](https://github.com/okta/okta-auth-js/blob/master/lib/myaccount/phoneApi.ts#L44)

___

### deleteEmail

▸ **deleteEmail**(`oktaAuth`, `options?`): `Promise`<[`BaseTransaction`](../classes/index.BaseTransaction.md)\>

**`scope:`** okta.myaccount.email.manage

#### Parameters

| Name | Type |
| :------ | :------ |
| `oktaAuth` | `OktaAuthInterface` |
| `options?` | [`MyAccountRequestOptions`](types.md#myaccountrequestoptions) |

#### Returns

`Promise`<[`BaseTransaction`](../classes/index.BaseTransaction.md)\>

#### Defined in

[emailApi.ts:63](https://github.com/okta/okta-auth-js/blob/master/lib/myaccount/emailApi.ts#L63)

___

### deletePhone

▸ **deletePhone**(`oktaAuth`, `options?`): `Promise`<[`BaseTransaction`](../classes/index.BaseTransaction.md)\>

**`scope:`** okta.myaccount.phone.manage

#### Parameters

| Name | Type |
| :------ | :------ |
| `oktaAuth` | `OktaAuthInterface` |
| `options?` | [`MyAccountRequestOptions`](types.md#myaccountrequestoptions) |

#### Returns

`Promise`<[`BaseTransaction`](../classes/index.BaseTransaction.md)\>

#### Defined in

[phoneApi.ts:62](https://github.com/okta/okta-auth-js/blob/master/lib/myaccount/phoneApi.ts#L62)

___

### getEmail

▸ **getEmail**(`oktaAuth`, `options?`): `Promise`<[`EmailTransaction`](../classes/index.EmailTransaction.md)\>

**`scope:`** okta.myaccount.email.read

#### Parameters

| Name | Type |
| :------ | :------ |
| `oktaAuth` | `OktaAuthInterface` |
| `options?` | [`MyAccountRequestOptions`](types.md#myaccountrequestoptions) |

#### Returns

`Promise`<[`EmailTransaction`](../classes/index.EmailTransaction.md)\>

#### Defined in

[emailApi.ts:28](https://github.com/okta/okta-auth-js/blob/master/lib/myaccount/emailApi.ts#L28)

___

### getEmailChallenge

▸ **getEmailChallenge**(`oktaAuth`, `options?`): `Promise`<[`EmailChallengeTransaction`](../classes/index.EmailChallengeTransaction.md)\>

**`scope:`** okta.myaccount.email.read

#### Parameters

| Name | Type |
| :------ | :------ |
| `oktaAuth` | `OktaAuthInterface` |
| `options?` | [`MyAccountRequestOptions`](types.md#myaccountrequestoptions) |

#### Returns

`Promise`<[`EmailChallengeTransaction`](../classes/index.EmailChallengeTransaction.md)\>

#### Defined in

[emailApi.ts:96](https://github.com/okta/okta-auth-js/blob/master/lib/myaccount/emailApi.ts#L96)

___

### getEmails

▸ **getEmails**(`oktaAuth`, `options?`): `Promise`<[`EmailTransaction`](../classes/index.EmailTransaction.md)[]\>

**`scope:`** okta.myaccount.email.read

#### Parameters

| Name | Type |
| :------ | :------ |
| `oktaAuth` | `OktaAuthInterface` |
| `options?` | [`MyAccountRequestOptions`](types.md#myaccountrequestoptions) |

#### Returns

`Promise`<[`EmailTransaction`](../classes/index.EmailTransaction.md)[]\>

#### Defined in

[emailApi.ts:12](https://github.com/okta/okta-auth-js/blob/master/lib/myaccount/emailApi.ts#L12)

___

### getPhone

▸ **getPhone**(`oktaAuth`, `options?`): `Promise`<[`PhoneTransaction`](../classes/index.PhoneTransaction.md)\>

**`scope:`** okta.myaccount.phone.read

#### Parameters

| Name | Type |
| :------ | :------ |
| `oktaAuth` | `OktaAuthInterface` |
| `options?` | [`MyAccountRequestOptions`](types.md#myaccountrequestoptions) |

#### Returns

`Promise`<[`PhoneTransaction`](../classes/index.PhoneTransaction.md)\>

#### Defined in

[phoneApi.ts:27](https://github.com/okta/okta-auth-js/blob/master/lib/myaccount/phoneApi.ts#L27)

___

### getPhones

▸ **getPhones**(`oktaAuth`, `options?`): `Promise`<[`PhoneTransaction`](../classes/index.PhoneTransaction.md)[]\>

**`scope:`** okta.myaccount.phone.read

#### Parameters

| Name | Type |
| :------ | :------ |
| `oktaAuth` | `OktaAuthInterface` |
| `options?` | [`MyAccountRequestOptions`](types.md#myaccountrequestoptions) |

#### Returns

`Promise`<[`PhoneTransaction`](../classes/index.PhoneTransaction.md)[]\>

#### Defined in

[phoneApi.ts:11](https://github.com/okta/okta-auth-js/blob/master/lib/myaccount/phoneApi.ts#L11)

___

### getProfile

▸ **getProfile**(`oktaAuth`, `options?`): `Promise`<[`ProfileTransaction`](../classes/index.ProfileTransaction.md)\>

**`scope:`** okta.myaccount.profile.read

#### Parameters

| Name | Type |
| :------ | :------ |
| `oktaAuth` | `OktaAuthInterface` |
| `options?` | [`MyAccountRequestOptions`](types.md#myaccountrequestoptions) |

#### Returns

`Promise`<[`ProfileTransaction`](../classes/index.ProfileTransaction.md)\>

#### Defined in

[profileApi.ts:11](https://github.com/okta/okta-auth-js/blob/master/lib/myaccount/profileApi.ts#L11)

___

### getProfileSchema

▸ **getProfileSchema**(`oktaAuth`, `options?`): `Promise`<[`ProfileSchemaTransaction`](../classes/index.ProfileSchemaTransaction.md)\>

**`scope:`** okta.myaccount.profile.read

#### Parameters

| Name | Type |
| :------ | :------ |
| `oktaAuth` | `OktaAuthInterface` |
| `options?` | [`MyAccountRequestOptions`](types.md#myaccountrequestoptions) |

#### Returns

`Promise`<[`ProfileSchemaTransaction`](../classes/index.ProfileSchemaTransaction.md)\>

#### Defined in

[profileApi.ts:42](https://github.com/okta/okta-auth-js/blob/master/lib/myaccount/profileApi.ts#L42)

___

### sendEmailChallenge

▸ **sendEmailChallenge**(`oktaAuth`, `options?`): `Promise`<[`EmailChallengeTransaction`](../classes/index.EmailChallengeTransaction.md)\>

**`scope:`** okta.myaccount.email.read

#### Parameters

| Name | Type |
| :------ | :------ |
| `oktaAuth` | `OktaAuthInterface` |
| `options?` | [`MyAccountRequestOptions`](types.md#myaccountrequestoptions) |

#### Returns

`Promise`<[`EmailChallengeTransaction`](../classes/index.EmailChallengeTransaction.md)\>

#### Defined in

[emailApi.ts:79](https://github.com/okta/okta-auth-js/blob/master/lib/myaccount/emailApi.ts#L79)

___

### sendPhoneChallenge

▸ **sendPhoneChallenge**(`oktaAuth`, `options?`): `Promise`<[`BaseTransaction`](../classes/index.BaseTransaction.md)\>

**`scope:`** okta.myaccount.phone.manage

#### Parameters

| Name | Type |
| :------ | :------ |
| `oktaAuth` | `OktaAuthInterface` |
| `options?` | [`MyAccountRequestOptions`](types.md#myaccountrequestoptions) |

#### Returns

`Promise`<[`BaseTransaction`](../classes/index.BaseTransaction.md)\>

#### Defined in

[phoneApi.ts:78](https://github.com/okta/okta-auth-js/blob/master/lib/myaccount/phoneApi.ts#L78)

___

### updateProfile

▸ **updateProfile**(`oktaAuth`, `options?`): `Promise`<[`ProfileTransaction`](../classes/index.ProfileTransaction.md)\>

**`scope:`** okta.myaccount.profile.manage

#### Parameters

| Name | Type |
| :------ | :------ |
| `oktaAuth` | `OktaAuthInterface` |
| `options?` | [`MyAccountRequestOptions`](types.md#myaccountrequestoptions) |

#### Returns

`Promise`<[`ProfileTransaction`](../classes/index.ProfileTransaction.md)\>

#### Defined in

[profileApi.ts:24](https://github.com/okta/okta-auth-js/blob/master/lib/myaccount/profileApi.ts#L24)

___

### verifyEmailChallenge

▸ **verifyEmailChallenge**(`oktaAuth`, `options?`): `Promise`<[`BaseTransaction`](../classes/index.BaseTransaction.md)\>

**`scope:`** okta.myaccount.email.manage

#### Parameters

| Name | Type |
| :------ | :------ |
| `oktaAuth` | `OktaAuthInterface` |
| `options?` | [`MyAccountRequestOptions`](types.md#myaccountrequestoptions) |

#### Returns

`Promise`<[`BaseTransaction`](../classes/index.BaseTransaction.md)\>

#### Defined in

[emailApi.ts:113](https://github.com/okta/okta-auth-js/blob/master/lib/myaccount/emailApi.ts#L113)

___

### verifyPhoneChallenge

▸ **verifyPhoneChallenge**(`oktaAuth`, `options?`): `Promise`<[`BaseTransaction`](../classes/index.BaseTransaction.md)\>

**`scope:`** okta.myaccount.phone.manage

#### Parameters

| Name | Type |
| :------ | :------ |
| `oktaAuth` | `OktaAuthInterface` |
| `options?` | [`MyAccountRequestOptions`](types.md#myaccountrequestoptions) |

#### Returns

`Promise`<[`BaseTransaction`](../classes/index.BaseTransaction.md)\>

#### Defined in

[phoneApi.ts:95](https://github.com/okta/okta-auth-js/blob/master/lib/myaccount/phoneApi.ts#L95)
