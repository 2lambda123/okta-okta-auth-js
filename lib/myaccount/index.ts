export * from './profileApi';
export * from './emailApi';
export * from './phoneApi';
export * from './types';
