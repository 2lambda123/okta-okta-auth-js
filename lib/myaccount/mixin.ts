import { OktaAuthConstructor } from '../base/types';
import {
  OAuthStorageManagerInterface,
  OAuthTransactionMeta,
  PKCETransactionMeta,
} from '../oidc/types';
import { OktaAuthCoreInterface, OktaAuthCoreOptions } from '../core/types';
import { OktaAuthMyAccountInterface } from './types';

import * as MyAccountMethods from './api';

export function mixinMyAccount
<
  M extends OAuthTransactionMeta = PKCETransactionMeta,
  S extends OAuthStorageManagerInterface<M> = OAuthStorageManagerInterface<M>,
  O extends OktaAuthCoreOptions<M, S> = OktaAuthCoreOptions<M, S>,
  TBase extends OktaAuthConstructor<O, OktaAuthCoreInterface<M, S, O>>
    = OktaAuthConstructor<O, OktaAuthCoreInterface<M, S, O>>
>
(Base: TBase): TBase & OktaAuthConstructor<O, OktaAuthMyAccountInterface<M, S, O>>
{
  return class OktaAuthMyAccount extends Base implements OktaAuthMyAccountInterface<M, S, O>
  {
    myaccount: any;
    
    constructor(...args: any[]) {
      super(...args);

      this.myaccount = Object.entries(MyAccountMethods)
        .filter(([ name ]) => name !== 'default')
        .reduce((acc, [name, fn]) => {
          acc[name] = (fn as any).bind(null, this);
          return acc;
        }, {});
    }
  };
}
