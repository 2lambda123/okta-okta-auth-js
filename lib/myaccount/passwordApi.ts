import { sendRequest } from './request';
import { 
  IAPIFunction,
  BaseTransaction, 
  PasswordTransaction
} from './types';

/**
 * @scope: okta.myAccount.password.read
 */
export const getPassword: IAPIFunction<PasswordTransaction> = async (
  oktaAuth,
  options
) => {
  const transaction = await sendRequest(oktaAuth, {
    url: `/idp/myaccount/password`,
    method: 'GET',
    accessToken: options?.accessToken,
    transactionClassName: 'PasswordTransaction'
  }) as PasswordTransaction;
  return transaction;
};

/**
 * @scope: okta.myAccount.password.manage
 */
// TODO: type check payload?
export const enrollPassword: IAPIFunction<PasswordTransaction> = async (
  oktaAuth, 
  options
): Promise<PasswordTransaction> => {
  const { accessToken, payload } = options!;
  const transaction = await sendRequest(oktaAuth, {
    url: '/idp/myaccount/password',
    method: 'POST',
    payload,
    accessToken,
    transactionClassName: 'PasswordTransaction'
  }) as PasswordTransaction;
  return transaction;
};

/**
 * @scope: okta.myAccount.password.manage
 */
export const updatePassword: IAPIFunction<PasswordTransaction> = async (
  oktaAuth, 
  options
): Promise<PasswordTransaction> => {
  const { accessToken, payload } = options!;
  const transaction = await sendRequest(oktaAuth, {
    url: '/idp/myaccount/password',
    method: 'PUT',
    payload,
    accessToken,
    transactionClassName: 'PasswordTransaction'
  }) as PasswordTransaction;
  return transaction;
};

/**
 * @scope: okta.myAccount.password.manage
 */
export const deletePassword: IAPIFunction<BaseTransaction> = async (
  oktaAuth, 
  options?
) => {
  const transaction = await sendRequest(oktaAuth, {
    url: `/idp/myaccount/password`,
    method: 'DELETE',
    accessToken: options?.accessToken,
  }) as BaseTransaction;
  return transaction;
};
