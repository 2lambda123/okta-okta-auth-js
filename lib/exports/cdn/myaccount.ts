import { OktaAuth } from '../myaccount';

// Export only a single object
export default OktaAuth;
