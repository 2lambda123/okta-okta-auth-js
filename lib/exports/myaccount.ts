import { createCoreOptionsConstructor } from '../core/options';
import { createCoreStorageManager } from '../core/storage';
import { createOktaAuthMyAccount } from '../myaccount/factory';
import {
  CoreStorageManagerInterface,
  OktaAuthCoreOptions,
  OktaAuthOptionsConstructor,
  StorageManagerConstructor
} from '../types';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface OktaAuthOptions extends OktaAuthCoreOptions {}

const OptionsConstructor: OktaAuthOptionsConstructor<OktaAuthOptions> = createCoreOptionsConstructor();
const StorageManager: StorageManagerConstructor<CoreStorageManagerInterface> = createCoreStorageManager();
const OktaAuthMyAccount = createOktaAuthMyAccount(StorageManager, OptionsConstructor);

class OktaAuth extends OktaAuthMyAccount {
  constructor(options: OktaAuthOptions) {
    super(options);
  }
}

export default OktaAuth;
export { OktaAuth };
export * from './common';
export * from '../myaccount';

