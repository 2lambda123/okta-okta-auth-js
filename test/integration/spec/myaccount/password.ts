import {
  getPassword,
  enrollPassword,
  updatePassword,
  deletePassword
} from '../../../../lib/myaccount';
import {
  createClient,
  signinAndGetTokens,
  signinAndGetTokensViaEmail
} from '../../util';
import { PasswordTransaction } from '../../../../lib/myaccount/transactions';

// TODO: generate dynamic profile with a18n api
describe('MyAccount Password API', () => {
  let client, token;

  beforeAll(async () => {
    client = createClient({});
    const {
      tokens: {
        accessToken: { accessToken } = {}
      }
    } = await signinAndGetTokens(client, {
      scopes: [
        'openid',
        'profile',
        'okta.myAccount.password.read',
        'okta.myAccount.password.manage'
      ]
    });
    token = accessToken!;

    try {
      const { status, tokens } = await signinAndGetTokensViaEmail(client);
      console.log('#####', status, tokens, '######');
    }
    catch (err) {
      console.log('#####', err, '######');
    }
  });

  describe('Manage with Transaction functions', () => {
    let password: PasswordTransaction;

    afterEach(async () => {
      if (password && password.delete) {
        try {
          await password.delete();
        } catch {
          // do nothing
        }
      }
    });

    it.only('can manage password with transaction functions', async () => {
      password = await enrollPassword(client, {
        accessToken: token,
        payload: {
          profile: {
            password: 'MyPassword'
          }
        }
      });
      expect(password).toMatchSnapshot({
        headers: expect.any(Object),
        id: expect.any(String),
        status: expect('ACTIVE')
      });

      // get password (status ACTIVE)
      let transaction = await password.get();
      expect(transaction).toMatchSnapshot({
        headers: expect.any(Object),
        id: expect.any(String),
        status: expect('ACTIVE')
      });

      // update password
      transaction = await password.update!({
        profile: {
          password: 'MyUpdatedPassword'
        }
      });
      expect(transaction).toMatchSnapshot({
        headers: expect.any(Object),
        id: expect.any(String),
        status: expect('ACTIVE')
      });

      // delete
      const deleteTransaction = await password.delete!();
      expect(deleteTransaction).toMatchSnapshot();

      // get password (status NOT_ENROLLED)
      transaction = await password.get();
      expect(transaction).toMatchSnapshot({
        headers: expect.any(Object),
        status: expect('NOT_ENROLLED')
      });

      transaction = await password.enroll!({
        profile: {
          password: 'MyNewlyEnrolledPassword'
        }
      });
      expect(transaction).toMatchSnapshot({
        headers: expect.any(Object),
        id: expect.any(String),
        status: expect('ACTIVE')
      });

      // clean up
      await password.delete!();
    });
  });

  describe('Manage with SDK methods', () => {
    let password: PasswordTransaction;

    afterEach(async () => {
      if (password && password.delete) {
        try {
          await password.delete();
        } catch {
          // do nothing
        }
      }
    });

    it('can manage password with SDK methods', async () => {
      // create test password
      password = await enrollPassword(client, {
        accessToken: token,
        payload: {
          profile: {
            password: 'MyPassword'
          },
        }
      });
      expect(password).toMatchSnapshot({
        headers: expect.any(Object),
        id: expect.any(String),
        status: expect('ACTIVE')
      });

      // get password
      password = await getPassword(client, { accessToken: token });
      expect(password).toMatchSnapshot({
        headers: expect.any(Object),
        id: expect.any(String),
        status: expect('ACTIVE')
      });

      password = await updatePassword(client, {
        accessToken: token,
        payload: {
          profile: {
            password: 'MyPassword'
          },
        }
      });
      expect(password).toMatchSnapshot({
        headers: expect.any(Object),
        id: expect.any(String),
        status: expect('ACTIVE')
      });

      // delete
      const deleteTransaction = await deletePassword(client, { accessToken: token });
      expect(deleteTransaction).toMatchSnapshot({
        headers: expect.any(Object),
        status: expect('NOT_ENROLLED')
      });
    });

  });

});
