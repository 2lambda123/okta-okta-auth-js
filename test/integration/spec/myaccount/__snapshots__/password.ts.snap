// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyAccount Password API Manage with SDK methods can manage password with SDK methods 1`] = `
Object {
  "created": "2022-04-20T17:02:41.000Z",
  "delete": Any<Function>,
  "get": Any<Function>,
  "headers": Any<Object>,
  "id": Any<String>,
  "lastUpdated": "2022-11-15T23:05:20.000Z",
  "status": "ACTIVE",
  "update": Any<Function>,
}
`;

exports[`MyAccount Password API Manage with SDK methods can manage password with SDK methods 2`] = `
Object {
  "created": "2022-04-20T17:02:41.000Z",
  "delete": Any<Function>,
  "get": Any<Function>,
  "headers": Any<Object>,
  "id": Any<String>,
  "lastUpdated": "2022-11-15T23:05:20.000Z",
  "status": "ACTIVE",
  "update": Any<Function>,
}
`;

exports[`MyAccount Password API Manage with SDK methods can manage password with SDK methods 3`] = `
Object {
  "created": "2022-04-20T17:02:41.000Z",
  "delete": Any<Function>,
  "get": Any<Function>,
  "headers": Any<Object>,
  "id": Any<String>,
  "lastUpdated": "2022-11-15T23:05:21.000Z",
  "status": "ACTIVE",
  "update": Any<Function>,
}
`;

exports[`MyAccount Password API Manage with SDK methods can manage password with SDK methods 4`] = `BaseTransaction {}`;

exports[`MyAccount Password API Manage with Transaction functions can manage password with transaction functions 1`] = `
Object {
  "created": "2022-04-20T17:02:41.000Z",
  "delete": Any<Function>,
  "get": Any<Function>,
  "headers": Any<Object>,
  "id": Any<String>,
  "lastUpdated": "2022-11-15T23:05:15.000Z",
  "status": "ACTIVE",
  "update": Any<Function>,
}
`;

exports[`MyAccount Password API Manage with Transaction functions can manage password with transaction functions 2`] = `
Object {
  "created": "2022-04-20T17:02:41.000Z",
  "delete": Any<Function>,
  "get": Any<Function>,
  "headers": Any<Object>,
  "id": Any<String>,
  "lastUpdated": "2022-11-15T23:05:15.000Z",
  "status": "ACTIVE",
  "update": Any<Function>,
}
`;

exports[`MyAccount Password API Manage with Transaction functions can manage password with transaction functions 3`] = `
Object {
  "created": "2022-04-20T17:02:41.000Z",
  "delete": Any<Function>,
  "get": Any<Function>,
  "headers": Any<Object>,
  "id": Any<String>,
  "lastUpdated": "2022-11-15T23:05:17.000Z",
  "status": "ACTIVE",
  "update": Any<Function>,
}
`;

exports[`MyAccount Password API Manage with Transaction functions can manage password with transaction functions 4`] = `BaseTransaction {}`;

exports[`MyAccount Password API Manage with Transaction functions can manage password with transaction functions 5`] = `
Object {
  "created": undefined,
  "enroll": Any<Function>,
  "headers": Any<Object>,
  "id": undefined,
  "lastUpdated": undefined,
  "status": "NOT_ENROLLED",
}
`;

exports[`MyAccount Password API Manage with Transaction functions can manage password with transaction functions 6`] = `
Object {
  "created": "2022-04-20T17:02:41.000Z",
  "delete": [Function],
  "get": [Function],
  "headers": Any<Object>,
  "id": Any<String>,
  "lastUpdated": "2022-11-15T23:05:18.000Z",
  "status": "ACTIVE",
  "update": [Function],
}
`;
